### PR TITLE
gokey: 0.1.2-unstable-2023-11-16 -> 0.1.3

### PR DESCRIPTION
**Pull request by con-f-use**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by con-f-use** [comment]
https://github.com/cloudflare/gokey/issues/75

File: pkgs/by-name/go/gokey/package.nix [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
+  version = "0.1.3";
+
+  patches = [ ./version.patch ];
 
   src = fetchFromGitHub {
     owner = "cloudflare";
     repo = "gokey";
-    rev = "26fcef24d123e0eaf7b92224e6880f529f94aa9f";
-    hash = "sha256-nt4fO8NKYfRkpoC1z8zDrEZC7+fo6sU/ZOHCMHIAT58=";
+    rev = "b5342a3cf158fa31ce4b6087955f7625b16b2031";
```
Comment (NickCao) [inline pkgs/by-name/go/gokey/package.nix:11]:
```suggestion
    rev = "refs/tags/${version}";
```

Comment (NickCao) [reply pkgs/by-name/go/gokey/package.nix:11]: The majority of packages in nixpkgs use the version tag, I suggest we just follow suit.

File: pkgs/by-name/go/gokey/package.nix [hunk 2]
```diff
@@ -1,16 +1,18 @@
 { lib, buildGoModule, fetchFromGitHub }:
 buildGoModule {
```
Comment (NickCao) [inline pkgs/by-name/go/gokey/package.nix:2]:
```suggestion
buildGoModule rec {
```

Missing a rec here, and please squash the commits.

Comment (con-f-use) [reply pkgs/by-name/go/gokey/package.nix]: Done. Thanks for the review.

File: pkgs/by-name/go/gokey/package.nix [hunk 3]
```diff
@@ -1,16 +1,18 @@
...
+  version = "0.1.3";
+
+  patches = [ ./version.patch ];
 
   src = fetchFromGitHub {
     owner = "cloudflare";
     repo = "gokey";
-    rev = "26fcef24d123e0eaf7b92224e6880f529f94aa9f";
-    hash = "sha256-nt4fO8NKYfRkpoC1z8zDrEZC7+fo6sU/ZOHCMHIAT58=";
+    rev = "refs/tags/${version}";
```
Comment (NickCao) [inline pkgs/by-name/go/gokey/package.nix:11]:
```suggestion
    rev = "refs/tags/v${version}";
```

My bad, the tag comes with a v prefix.

File: pkgs/by-name/go/gokey/package.nix [hunk 4]
```diff
@@ -1,16 +1,18 @@
 { lib, buildGoModule, fetchFromGitHub }:
-buildGoModule {
+buildGoModule rec {
   pname = "gokey";
-  version = "0.1.2-unstable-2023-11-16";
+  version = "0.1.3";
+
+  patches = [ ./version.patch ];
```
Comment (NickCao) [inline pkgs/by-name/go/gokey/package.nix:6]: Try to upstream this patch if you are willing to.

**Review by NickCao — COMMENTED** [review]
Apart from the v prefix, LGTM.

File: pkgs/by-name/go/gokey/package.nix (hunk 3 above)
Comment (con-f-use) [reply pkgs/by-name/go/gokey/package.nix]: Hmm... weird, it did build for me even without the `v`, but it's added now.

Comment (NickCao) [reply pkgs/by-name/go/gokey/package.nix]: That's a footgun with FOD (fixed output derivations), as long as the hash doesn't change, it never get rebuilt.

**Review by NickCao — APPROVED** [review]
Waiting for ofborg.

**Pull request merged**